src/datastore/src/main/java/org/locationtech/geogig/geotools/data/GeogigFeatureSource.java

```diff
@@ -247,6 +257,51 @@ protected int getCountInternal(Query query) throws IOException {
     protected FeatureReader<SimpleFeatureType, SimpleFeature> getReaderInternal(final Query query)
             throws IOException {
 
+        FeatureReader<SimpleFeatureType, SimpleFeature> featureReader;
+        featureReader = getNativeReader(query, true);
+
+        // do retyping to satisfy the Query's requested set of properties in the requested order
```

Comment:
I think the spec indicates that the order isn't important in the request - however, I think the result needs the properties in the correct (full feature type) order.

Not quite sure of the exact schematic, but if the featuretype has the properties in a specific order then the property-reduced result should have them in the same order (i.e. the property-reduced results is the full-result with the un-needed properties removed).  I don't think re-ordering is correct.  However, I might be wrong.